src/components/Checkbox/Checkbox.js

```diff
@@ -12,7 +12,7 @@ import { settings } from 'carbon-components';
 
 const { prefix } = settings;
 
-const Checkbox = ({
+const CheckboxImpl = ({
```

Comment:
Though merging `Checkbox` and `CheckboxImpl` prevents `<Checkbox><CheckboxImpl ... /><Checkbox>` being shown in React DevTools, such technique cannot be applied to stateful components, and thus I kept it intact.

Reply:
would this be a reason not to convert `<Checkbox>` to a stateful component? I saw that you suggested that as a possible alternative solution to set indeterminate prop when a ref is also passed into `<Checkbox>`

Reply:
@emyarod Agreed that `<...Impl>` showing up in React DevTools (and other env) is noisy, but seems that we cannot avoid that e.g. for `<NumberInput>`, which is stateful. I think this shouldn't stop us from moving `<Checkbox>` to stateful if we think we need it.

Reply:
Talked to @emyarod offline, he seems fine with `ForwardRef(Checkbox)` though he seems to prefer the other. I went ahead with `ForwardRef(Checkbox)` based on that.